src/external_signer.cpp

```diff
@@ -75,15 +76,14 @@ bool ExternalSigner::SignTransaction(PartiallySignedTransaction& psbtx, std::str
     ssTx << psbtx;
 
     // Check if signer fingerprint matches any input master key fingerprint
-    bool match = false;
-    for (unsigned int i = 0; i < psbtx.inputs.size(); ++i) {
-        const PSBTInput& input = psbtx.inputs[i];
+    auto matches_signer_fingerprint = [&](const PSBTInput& input) {
         for (const auto& entry : input.hd_keypaths) {
-            if (m_fingerprint == strprintf("%08x", ReadBE32(entry.second.fingerprint))) match = true;
+            if (m_fingerprint == strprintf("%08x", ReadBE32(entry.second.fingerprint))) return true;
         }
-    }
+        return false;
+    };
```

Comment:
I'm curious, could you get rid of the other for loop as well with the same approach? Would that be cleaner? Something like:

```c++
return std::any_of(input.hd_keypaths.begin(), input.hd_keypaths.end(), [](auto& entry){ return m_fingerprint == strprintf("%08x", ReadBE32(entry.second.fingerprint)); });
```